packages/copyable/src/Copyable.tsx

```diff
@@ -0,0 +1,283 @@
+  font-size: 16px;
+`;
+
+const buttonWrapperStyle = css`
+  position: absolute;
+  z-index: 1;
+  right: 0;
+  top: 0;
+  height: 100%;
+`;
+
+const buttonStyle = css`
+  height: 100%;
+  border-top-left-radius: 0;
+  border-bottom-left-radius: 0;
+  filter: drop-shadow(2px 0 6px ${transparentize(0.7, uiColors.gray.dark1)});
+`;
+
+const iconStyle = css`
+  padding-right: 6px;
+`;
+
+interface CopyableProps {
+  variant?: Variant;
+  children: string;
+  label: string;
+  description?: string;
+  copyable?: boolean;
+  size?: Size;
+}
+
+export default function Copyable({
+  variant = Variant.Default,
+  children,
+  label,
+  description,
+  copyable = true,
+  size = Size.Default,
+}: CopyableProps) {
+  const colorSet = colorSets[variant];
+
+  const [copied, setCopied] = useState(false);
+  const [showCopyButton, setShowCopyButton] = useState(false);
+  const [buttonRef, setButtonRef] = useState<HTMLButtonElement>();
+
+  useEffect(() => {
+    setShowCopyButton(copyable && ClipboardJS.isSupported());
+  }, [copyable]);
+
+  let copyButton: JSX.Element | undefined;
+
+  if (showCopyButton) {
+    let buttonVariant: ButtonVariant;
+
+    switch (variant) {
+      case Variant.Default:
+        buttonVariant = ButtonVariant.Default;
+        break;
+      case Variant.Dark:
+        buttonVariant = ButtonVariant.Dark;
+        break;
+    }
+
+    const trigger = (
+      <Button
+        ref={setButtonRef}
+        variant={buttonVariant}
+        className={buttonStyle}
+        onClick={() => setCopied(true)}
+      >
+        <CopyIcon size="large" className={iconStyle} />
+        Copy
+      </Button>
+    );
+
+    let tooltipVariant: TooltipVariant;
+
+    switch (variant) {
+      case Variant.Default:
+        tooltipVariant = TooltipVariant.Dark;
+        break;
+      case Variant.Dark:
+        tooltipVariant = TooltipVariant.Light;
+        break;
+    }
+
+    copyButton = (
+      // `Tooltip` forces its trigger to be `position: relative` :/
+      <div className={buttonWrapperStyle}>
+        <Tooltip
+          open={copied}
+          variant={tooltipVariant}
+          align={Align.Top}
+          justify={Justify.Middle}
+          trigger={trigger}
+          triggerEvent={TriggerEvent.Click}
+        >
+          Copied!
+        </Tooltip>
+      </div>
+    );
+  }
+
+  useEffect(() => {
+    if (!buttonRef) {
+      return;
+    }
+
+    const clipboard = new ClipboardJS(buttonRef, {
+      text: () => children,
+    });
+
+    if (copied) {
+      let destroyed = false;
+
+      const timeoutId = setTimeout(() => {
+        setCopied(false);
+        clipboard.destroy();
+        destroyed = true;
+      }, 1500);
+
+      return () => {
+        clearTimeout(timeoutId);
+        if (!destroyed) {
+          clipboard.destroy();
+        }
+      };
+    }
+  }, [buttonRef, children, copied]);
+
+  return (
+    <>
+      {label && (
+        <div className={cx(labelStyle, css(`color: ${colorSet.label};`))}>
```

Comment:
I might be missing something but I think you can inline this css`` without the parens